src/azul/drs.py

```diff
@@ -269,21 +266,30 @@ def get_object(self,
         depends on the access method specified.
         """
         if access_id is None:
-            return self._get_object(object_id, access_method=access_method)
+            return self._get_object(drs_uri, access_method)
         else:
-            return self._get_object_access(object_id, access_id, access_method=access_method)
-
-    def _get_object(self, object_id: str, access_method: AccessMethod) -> str:
-        url = self.url / object_id
+            return self._get_object_access(drs_uri, access_id, access_method)
+
+    def _get_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
```

Comment:
I was just reading the DRS spec, and I realized that even with your improvements, we're still only support one type of DRS URI---we're not supporting https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#_compact_identifier_based_drs_uris.

I think we should leave this unsupported for now, but we _may_ want to add a comment or error message if the compact identifier based URIs are used.

Feel free to leave this out though, since our DRS client is just for internal use anyway.